keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/artifacts/PublishedArtifact.kt

```diff
@@ -17,5 +19,40 @@ data class PublishedArtifact(
   val artifactAccount: String? = null,
   val provenance: String? = null,
   val uuid: String? = null,
-  val metadata: Map<String, Any?> = emptyMap()
-)
+  val metadata: Map<String, Any?> = emptyMap(),
+  // keel-specific fields ---
+  val gitMetadata: GitMetadata? = null,
+  val buildMetadata: BuildMetadata? = null
+) {
+
+  // The stuff that matters to keel
+  constructor(
+    name: String,
+    type: String,
+    version: String,
+    status: ArtifactStatus? = null,
+    createdAt: Instant? = null,
+    gitMetadata: GitMetadata? = null,
+    buildMetadata: BuildMetadata? = null
+  ) : this(
+    name = name,
+    type = type.toLowerCase(),
+    reference = name,
+    version = version,
+    metadata = mapOf(
+      "releaseStatus" to status?.name,
+      "createdAt" to createdAt
+    ),
+    gitMetadata = gitMetadata,
+    buildMetadata = buildMetadata
+  )
+
+  val status: ArtifactStatus? = metadata["releaseStatus"]?.toString()
+    ?.let { ArtifactStatus.valueOf(it) }
+
+  val createdAt = (metadata["createdAt"] as? Long)
+    ?.let { Instant.ofEpochMilli(it) }
```

Comment:
should this be an instant, or the localdatetime format that we use in other tables?

Reply:
We use `Instant` in all the models. `LocalDateTime` is only used in the JOOQ metamodels.

Reply:
Ah okay, it's only if there is a timestamp column? Just curious, because you mentioned adding it to a column in a future pr

Reply:
Yeah, if the value is not present in the `metadata` map (which I plan to fill from the column in a future PR), then it will just get set to `null`. As for using a `Long` here, it's because that's what we'll get from Rocket  (where we use this same class to represent the published artifacts). We can still store it in a `TIMESTAMP` column in the DB, though.